### PR TITLE
Add unit test coverage for DetailsVideo.jsx

### PR DESCRIPTION
**Pull request by simiadebowale**
## Summary
Added unit test coverage for `DetailsVideo.jsx`.

## Related issue(s)

- https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/67660

## Testing done
Checked file coverage and confirmed it was 100% covered.

Screenshot 2023-10-16 at 11 16 44 PM

## Acceptance criteria
- [x] file coverage for Branches, functions, lines and statements is 100%

### CODE REVIEW

**Review by ryanshaw — APPROVED** [review]
@simiadebowale Took a look at this yesterday. I like this approach, it makes our unit testing more like unit testing and will probably increase our forward momentum. We should also consider removing the old unit tests after we have sufficient coverage with the new approach.

**Comment by vbahinwillit** [comment]
Seems like more tests are needed to give a level of confidence that the component works as expected. For example, this is a composite component with all of the inner components depending on the appointment as input. I would expect test cases verifying all of the inner components display correctly. I didn't check but this is also required for code coverage of this component. This may lead refactoring components since all the data in the appointment object is not necessary to display the inner components.

Technical Issue
The tests as written carry over state changes from test to test. Ex.

    appointment.videoData = {
      isVideo: true,
      isAtlas: false,
      extension: { patientHasMobileGfe: true },
      kind: 'MOBILE_ANY',
    };

modifies the global defined appointment object which is used in each test.

**Comment by simiadebowale** [comment]
@vbahinwillit - A bett

> Seems like more tests are needed to give a level of confidence that the component works as expected. For example, this is a composite component with all of the inner components depending on the appointment as input. I would expect test cases verifying all of the inner components display correctly. I didn't check but this is also required for code coverage of this component. This may lead refactoring components since all the data in the appointment object is not necessary to display the inner components.
> 
> Technical Issue The tests as written carry over state changes from test to test. Ex.
> 
> ```
> appointment.videoData = {
>   isVideo: true,
>   isAtlas: false,
>   extension: { patientHasMobileGfe: true },
>   kind: 'MOBILE_ANY',
> };
> ```
> 
> modifies the global defined appointment object which is used in each test.

@vbahinwillit - I think a better approach with unit tests in testing all child components separately. I.E, I would not expect tests for the `VideoLocation` or `VideoVisitProvider` components here. we have no logic affecting what is rendered by those components. Unit tests for `VideoLocation` or `VideoVisitProvider` should cover that logic.

**Review by jenniemc — APPROVED** [review]
LGTM!

**Comment by simiadebowale** [comment]
> Seems like more tests are needed to give a level of confidence that the component works as expected. For example, this is a composite component with all of the inner components depending on the appointment as input. I would expect test cases verifying all of the inner components display correctly. I didn't check but this is also required for code coverage of this component. This may lead refactoring components since all the data in the appointment object is not necessary to display the inner components.
> 
> Technical Issue The tests as written carry over state changes from test to test. Ex.
> 
> ```
> appointment.videoData = {
>   isVideo: true,
>   isAtlas: false,
>   extension: { patientHasMobileGfe: true },
>   kind: 'MOBILE_ANY',
> };
> ```
> 
> modifies the global defined appointment object which is used in each test.

@vbahinwillit - added checks for inner components & refactored appointment object.

**Review by vbahinwillit — APPROVED** [review]
LGTM. I wish there was a better way to setup the mock data. One of the things that I noticed with our tests is that there are some common configurations of mock data for appointments, facilities, etc. that are used through out the tests (cut and pasted). These are fixtures. It would be nice to do something like:

```
const atlasAppt = new AtlasAppointment();
const facility = new Facility('983');
...
```
Thoughts???

**Comment by simiadebowale** [comment]
> LGTM. I wish there was a better way to setup the mock data. One of the things that I noticed with our tests is that there are some common configurations of mock data for appointments, facilities, etc. that are used through out the tests (cut and pasted). These are fixtures. It would be nice to do something like:
> 
> ```
> const atlasAppt = new AtlasAppointment();
> const facility = new Facility('983');
> ...
> ```
> 
> Thoughts???

@vbahinwillit - I have created mocks for atlas appts & Facilities, we can continue to up date these helpers as we create more unit tests. I created a new file because I didn't want to mix them with our old mocks since we'll likely clean those up too

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx [hunk 1]
```diff
@@ -92,6 +92,52 @@ export default function DetailsVideo({ appointment, facilityData }) {
 }
 
 DetailsVideo.propTypes = {
-  appointment: PropTypes.object.isRequired,
-  facilityData: PropTypes.object,
+  appointment: PropTypes.shape({
```
Comment (simiadebowale) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx:95]: made Type checking more detailed for appt and facility props. Makes it easier to catch bugs with props...

**Pull request merged**